// @angular/core/src/application_module.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register(["@angular/core/src/linker/ng_module_factory", "@angular/core/src/application_module", "@angular/core/src/application_init", "@angular/core/src/application_ref", "@angular/core/src/linker/compiler", "@angular/core/src/animation/animation_queue", "@angular/core/src/linker/view_utils", "@angular/core/src/application_tokens", "@angular/core/src/zone/ng_zone", "@angular/core/src/render/api", "@angular/core/src/security", "@angular/core/src/i18n/tokens", "@angular/core/src/console", "@angular/core/src/error_handler", "@angular/core/src/testability/testability", "@angular/core/src/change_detection/differs/iterable_differs", "@angular/core/src/change_detection/differs/keyvalue_differs"], function (exports_1, context_1) {
    "use strict";
    var __extends = (this && this.__extends) || (function () {
        var extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return function (d, b) {
            extendStatics(d, b);
            function __() { this.constructor = d; }
            d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
        };
    })();
    var __moduleName = context_1 && context_1.id;
    var import0, import1, import2, import3, import4, import5, import6, import8, import9, import10, import11, import12, import13, import14, import15, import16, import17, ApplicationModuleInjector, ApplicationModuleNgFactory;
    return {
        setters: [
            function (import0_1) {
                import0 = import0_1;
            },
            function (import1_1) {
                import1 = import1_1;
            },
            function (import2_1) {
                import2 = import2_1;
            },
            function (import3_1) {
                import3 = import3_1;
            },
            function (import4_1) {
                import4 = import4_1;
            },
            function (import5_1) {
                import5 = import5_1;
            },
            function (import6_1) {
                import6 = import6_1;
            },
            function (import8_1) {
                import8 = import8_1;
            },
            function (import9_1) {
                import9 = import9_1;
            },
            function (import10_1) {
                import10 = import10_1;
            },
            function (import11_1) {
                import11 = import11_1;
            },
            function (import12_1) {
                import12 = import12_1;
            },
            function (import13_1) {
                import13 = import13_1;
            },
            function (import14_1) {
                import14 = import14_1;
            },
            function (import15_1) {
                import15 = import15_1;
            },
            function (import16_1) {
                import16 = import16_1;
            },
            function (import17_1) {
                import17 = import17_1;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            ApplicationModuleInjector = (function (_super) {
                __extends(ApplicationModuleInjector, _super);
                function ApplicationModuleInjector(parent) {
                    return _super.call(this, parent, [], []) || this;
                }
                Object.defineProperty(ApplicationModuleInjector.prototype, "_ApplicationRef_3", {
                    get: function () {
                        if ((this.__ApplicationRef_3 == null)) {
                            (this.__ApplicationRef_3 = this._ApplicationRef__2);
                        }
                        return this.__ApplicationRef_3;
                    },
                    enumerable: true,
                    configurable: true
                });
                Object.defineProperty(ApplicationModuleInjector.prototype, "_Compiler_4", {
                    get: function () {
                        if ((this.__Compiler_4 == null)) {
                            (this.__Compiler_4 = new import4.Compiler());
                        }
                        return this.__Compiler_4;
                    },
                    enumerable: true,
                    configurable: true
                });
                Object.defineProperty(ApplicationModuleInjector.prototype, "_APP_ID_5", {
                    get: function () {
                        if ((this.__APP_ID_5 == null)) {
                            (this.__APP_ID_5 = import8._appIdRandomProviderFactory());
                        }
                        return this.__APP_ID_5;
                    },
                    enumerable: true,
                    configurable: true
                });
                Object.defineProperty(ApplicationModuleInjector.prototype, "_AnimationQueue_6", {
                    get: function () {
                        if ((this.__AnimationQueue_6 == null)) {
                            (this.__AnimationQueue_6 = new import5.AnimationQueue(this.parent.get(import9.NgZone)));
                        }
                        return this.__AnimationQueue_6;
                    },
                    enumerable: true,
                    configurable: true
                });
                Object.defineProperty(ApplicationModuleInjector.prototype, "_ViewUtils_7", {
                    get: function () {
                        if ((this.__ViewUtils_7 == null)) {
                            (this.__ViewUtils_7 = new import6.ViewUtils(this.parent.get(import10.RootRenderer), this.parent.get(import11.Sanitizer), this._AnimationQueue_6));
                        }
                        return this.__ViewUtils_7;
                    },
                    enumerable: true,
                    configurable: true
                });
                Object.defineProperty(ApplicationModuleInjector.prototype, "_IterableDiffers_8", {
                    get: function () {
                        if ((this.__IterableDiffers_8 == null)) {
                            (this.__IterableDiffers_8 = import1._iterableDiffersFactory());
                        }
                        return this.__IterableDiffers_8;
                    },
                    enumerable: true,
                    configurable: true
                });
                Object.defineProperty(ApplicationModuleInjector.prototype, "_KeyValueDiffers_9", {
                    get: function () {
                        if ((this.__KeyValueDiffers_9 == null)) {
                            (this.__KeyValueDiffers_9 = import1._keyValueDiffersFactory());
                        }
                        return this.__KeyValueDiffers_9;
                    },
                    enumerable: true,
                    configurable: true
                });
                Object.defineProperty(ApplicationModuleInjector.prototype, "_LOCALE_ID_10", {
                    get: function () {
                        if ((this.__LOCALE_ID_10 == null)) {
                            (this.__LOCALE_ID_10 = import1._localeFactory(this.parent.get(import12.LOCALE_ID, null)));
                        }
                        return this.__LOCALE_ID_10;
                    },
                    enumerable: true,
                    configurable: true
                });
                ApplicationModuleInjector.prototype.createInternal = function () {
                    this._ApplicationModule_0 = new import1.ApplicationModule();
                    this._ApplicationInitStatus_1 = new import2.ApplicationInitStatus(this.parent.get(import2.APP_INITIALIZER, null));
                    this._ApplicationRef__2 = new import3.ApplicationRef_(this.parent.get(import9.NgZone), this.parent.get(import13.Console), this, this.parent.get(import14.ErrorHandler), this, this._ApplicationInitStatus_1, this.parent.get(import15.TestabilityRegistry, null), this.parent.get(import15.Testability, null));
                    return this._ApplicationModule_0;
                };
                ApplicationModuleInjector.prototype.getInternal = function (token, notFoundResult) {
                    if ((token === import1.ApplicationModule)) {
                        return this._ApplicationModule_0;
                    }
                    if ((token === import2.ApplicationInitStatus)) {
                        return this._ApplicationInitStatus_1;
                    }
                    if ((token === import3.ApplicationRef_)) {
                        return this._ApplicationRef__2;
                    }
                    if ((token === import3.ApplicationRef)) {
                        return this._ApplicationRef_3;
                    }
                    if ((token === import4.Compiler)) {
                        return this._Compiler_4;
                    }
                    if ((token === import8.APP_ID)) {
                        return this._APP_ID_5;
                    }
                    if ((token === import5.AnimationQueue)) {
                        return this._AnimationQueue_6;
                    }
                    if ((token === import6.ViewUtils)) {
                        return this._ViewUtils_7;
                    }
                    if ((token === import16.IterableDiffers)) {
                        return this._IterableDiffers_8;
                    }
                    if ((token === import17.KeyValueDiffers)) {
                        return this._KeyValueDiffers_9;
                    }
                    if ((token === import12.LOCALE_ID)) {
                        return this._LOCALE_ID_10;
                    }
                    return notFoundResult;
                };
                ApplicationModuleInjector.prototype.destroyInternal = function () {
                    this._ApplicationRef__2.ngOnDestroy();
                };
                return ApplicationModuleInjector;
            }(import0.NgModuleInjector));
            exports_1("ApplicationModuleNgFactory", ApplicationModuleNgFactory = new import0.NgModuleFactory(ApplicationModuleInjector, import1.ApplicationModule));
        }
    };
});
