// @angular/forms/src/directives/reactive_directives/form_control_name.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register(["@angular/forms/src/directives/reactive_directives/form_control_name", "@angular/core/src/change_detection/change_detection_util", "@angular/core/src/linker/view_utils"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var import0, import1, import3, Wrapper_FormControlName;
    return {
        setters: [
            function (import0_1) {
                import0 = import0_1;
            },
            function (import1_1) {
                import1 = import1_1;
            },
            function (import3_1) {
                import3 = import3_1;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_FormControlName = (function () {
                function Wrapper_FormControlName(p0, p1, p2, p3) {
                    this._changed = false;
                    this._changes = {};
                    this.context = new import0.FormControlName(p0, p1, p2, p3);
                    this._expr_0 = import1.UNINITIALIZED;
                    this._expr_1 = import1.UNINITIALIZED;
                    this._expr_2 = import1.UNINITIALIZED;
                }
                Wrapper_FormControlName.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_FormControlName.prototype.ngOnDestroy = function () {
                    this.context.ngOnDestroy();
                    (this.subscription0 && this.subscription0.unsubscribe());
                };
                Wrapper_FormControlName.prototype.check_name = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.name = currValue;
                        this._changes['name'] = new import1.SimpleChange(this._expr_0, currValue);
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_FormControlName.prototype.check_model = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
                        this._changed = true;
                        this.context.model = currValue;
                        this._changes['model'] = new import1.SimpleChange(this._expr_1, currValue);
                        this._expr_1 = currValue;
                    }
                };
                Wrapper_FormControlName.prototype.check_isDisabled = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
                        this._changed = true;
                        this.context.isDisabled = currValue;
                        this._changes['isDisabled'] = new import1.SimpleChange(this._expr_2, currValue);
                        this._expr_2 = currValue;
                    }
                };
                Wrapper_FormControlName.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    if (!throwOnChange) {
                        if (changed) {
                            this.context.ngOnChanges(this._changes);
                            this._changes = {};
                        }
                    }
                    return changed;
                };
                Wrapper_FormControlName.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_FormControlName.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_FormControlName.prototype.subscribe = function (view, _eventHandler, emit0) {
                    this._eventHandler = _eventHandler;
                    if (emit0) {
                        (this.subscription0 = this.context.update.subscribe(_eventHandler.bind(view, 'ngModelChange')));
                    }
                };
                return Wrapper_FormControlName;
            }());
            exports_1("Wrapper_FormControlName", Wrapper_FormControlName);
        }
    };
});
