// @angular/common/src/directives/ng_plural.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register(["@angular/common/src/directives/ng_plural", "@angular/core/src/change_detection/change_detection_util", "@angular/core/src/linker/view_utils"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var import0, import1, import3, Wrapper_NgPlural, Wrapper_NgPluralCase;
    return {
        setters: [
            function (import0_1) {
                import0 = import0_1;
            },
            function (import1_1) {
                import1 = import1_1;
            },
            function (import3_1) {
                import3 = import3_1;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_NgPlural = (function () {
                function Wrapper_NgPlural(p0) {
                    this._changed = false;
                    this.context = new import0.NgPlural(p0);
                    this._expr_0 = import1.UNINITIALIZED;
                }
                Wrapper_NgPlural.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_NgPlural.prototype.ngOnDestroy = function () {
                };
                Wrapper_NgPlural.prototype.check_ngPlural = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.ngPlural = currValue;
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_NgPlural.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_NgPlural.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_NgPlural.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_NgPlural.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_NgPlural;
            }());
            exports_1("Wrapper_NgPlural", Wrapper_NgPlural);
            Wrapper_NgPluralCase = (function () {
                function Wrapper_NgPluralCase(p0, p1, p2, p3) {
                    this._changed = false;
                    this.context = new import0.NgPluralCase(p0, p1, p2, p3);
                }
                Wrapper_NgPluralCase.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_NgPluralCase.prototype.ngOnDestroy = function () {
                };
                Wrapper_NgPluralCase.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_NgPluralCase.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_NgPluralCase.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_NgPluralCase.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_NgPluralCase;
            }());
            exports_1("Wrapper_NgPluralCase", Wrapper_NgPluralCase);
        }
    };
});
