// @angular/common/src/directives/ng_switch.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register(["@angular/common/src/directives/ng_switch", "@angular/core/src/change_detection/change_detection_util", "@angular/core/src/linker/view_utils"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var import0, import1, import3, Wrapper_NgSwitch, Wrapper_NgSwitchCase, Wrapper_NgSwitchDefault;
    return {
        setters: [
            function (import0_1) {
                import0 = import0_1;
            },
            function (import1_1) {
                import1 = import1_1;
            },
            function (import3_1) {
                import3 = import3_1;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_NgSwitch = (function () {
                function Wrapper_NgSwitch() {
                    this._changed = false;
                    this.context = new import0.NgSwitch();
                    this._expr_0 = import1.UNINITIALIZED;
                }
                Wrapper_NgSwitch.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_NgSwitch.prototype.ngOnDestroy = function () {
                };
                Wrapper_NgSwitch.prototype.check_ngSwitch = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.ngSwitch = currValue;
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_NgSwitch.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_NgSwitch.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_NgSwitch.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_NgSwitch.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_NgSwitch;
            }());
            exports_1("Wrapper_NgSwitch", Wrapper_NgSwitch);
            Wrapper_NgSwitchCase = (function () {
                function Wrapper_NgSwitchCase(p0, p1, p2) {
                    this._changed = false;
                    this.context = new import0.NgSwitchCase(p0, p1, p2);
                    this._expr_0 = import1.UNINITIALIZED;
                }
                Wrapper_NgSwitchCase.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_NgSwitchCase.prototype.ngOnDestroy = function () {
                };
                Wrapper_NgSwitchCase.prototype.check_ngSwitchCase = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.ngSwitchCase = currValue;
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_NgSwitchCase.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    if (!throwOnChange) {
                        this.context.ngDoCheck();
                    }
                    return changed;
                };
                Wrapper_NgSwitchCase.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_NgSwitchCase.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_NgSwitchCase.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_NgSwitchCase;
            }());
            exports_1("Wrapper_NgSwitchCase", Wrapper_NgSwitchCase);
            Wrapper_NgSwitchDefault = (function () {
                function Wrapper_NgSwitchDefault(p0, p1, p2) {
                    this._changed = false;
                    this.context = new import0.NgSwitchDefault(p0, p1, p2);
                }
                Wrapper_NgSwitchDefault.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_NgSwitchDefault.prototype.ngOnDestroy = function () {
                };
                Wrapper_NgSwitchDefault.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_NgSwitchDefault.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_NgSwitchDefault.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_NgSwitchDefault.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_NgSwitchDefault;
            }());
            exports_1("Wrapper_NgSwitchDefault", Wrapper_NgSwitchDefault);
        }
    };
});
