// @angular/common/src/directives/ng_style.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register(["@angular/common/src/directives/ng_style", "@angular/core/src/change_detection/change_detection_util", "@angular/core/src/linker/view_utils"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var import0, import1, import3, Wrapper_NgStyle;
    return {
        setters: [
            function (import0_1) {
                import0 = import0_1;
            },
            function (import1_1) {
                import1 = import1_1;
            },
            function (import3_1) {
                import3 = import3_1;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_NgStyle = (function () {
                function Wrapper_NgStyle(p0, p1, p2) {
                    this._changed = false;
                    this.context = new import0.NgStyle(p0, p1, p2);
                    this._expr_0 = import1.UNINITIALIZED;
                }
                Wrapper_NgStyle.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_NgStyle.prototype.ngOnDestroy = function () {
                };
                Wrapper_NgStyle.prototype.check_ngStyle = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.ngStyle = currValue;
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_NgStyle.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    if (!throwOnChange) {
                        this.context.ngDoCheck();
                    }
                    return changed;
                };
                Wrapper_NgStyle.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_NgStyle.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_NgStyle.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_NgStyle;
            }());
            exports_1("Wrapper_NgStyle", Wrapper_NgStyle);
        }
    };
});
