// @angular/forms/src/directives/select_control_value_accessor.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register(["@angular/forms/src/directives/select_control_value_accessor", "@angular/core/src/change_detection/change_detection_util", "@angular/core/src/linker/view_utils"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var import0, import1, import3, Wrapper_NgSelectOption, Wrapper_SelectControlValueAccessor;
    return {
        setters: [
            function (import0_1) {
                import0 = import0_1;
            },
            function (import1_1) {
                import1 = import1_1;
            },
            function (import3_1) {
                import3 = import3_1;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_NgSelectOption = (function () {
                function Wrapper_NgSelectOption(p0, p1, p2) {
                    this._changed = false;
                    this.context = new import0.NgSelectOption(p0, p1, p2);
                    this._expr_0 = import1.UNINITIALIZED;
                    this._expr_1 = import1.UNINITIALIZED;
                }
                Wrapper_NgSelectOption.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_NgSelectOption.prototype.ngOnDestroy = function () {
                    this.context.ngOnDestroy();
                };
                Wrapper_NgSelectOption.prototype.check_ngValue = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.ngValue = currValue;
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_NgSelectOption.prototype.check_value = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
                        this._changed = true;
                        this.context.value = currValue;
                        this._expr_1 = currValue;
                    }
                };
                Wrapper_NgSelectOption.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_NgSelectOption.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_NgSelectOption.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_NgSelectOption.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_NgSelectOption;
            }());
            exports_1("Wrapper_NgSelectOption", Wrapper_NgSelectOption);
            Wrapper_SelectControlValueAccessor = (function () {
                function Wrapper_SelectControlValueAccessor(p0, p1) {
                    this._changed = false;
                    this.context = new import0.SelectControlValueAccessor(p0, p1);
                }
                Wrapper_SelectControlValueAccessor.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_SelectControlValueAccessor.prototype.ngOnDestroy = function () {
                };
                Wrapper_SelectControlValueAccessor.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_SelectControlValueAccessor.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_SelectControlValueAccessor.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    if ((eventName == 'change')) {
                        var pd_sub_0 = (this.context.onChange($event.target.value) !== false);
                        result = (pd_sub_0 && result);
                    }
                    if ((eventName == 'blur')) {
                        var pd_sub_1 = (this.context.onTouched() !== false);
                        result = (pd_sub_1 && result);
                    }
                    return result;
                };
                Wrapper_SelectControlValueAccessor.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_SelectControlValueAccessor;
            }());
            exports_1("Wrapper_SelectControlValueAccessor", Wrapper_SelectControlValueAccessor);
        }
    };
});
