// @angular/common/src/common_module.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register(["@angular/core/src/linker/ng_module_factory", "@angular/common/src/common_module", "@angular/common/src/localization", "@angular/core/src/i18n/tokens"], function (exports_1, context_1) {
    "use strict";
    var __extends = (this && this.__extends) || (function () {
        var extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return function (d, b) {
            extendStatics(d, b);
            function __() { this.constructor = d; }
            d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
        };
    })();
    var __moduleName = context_1 && context_1.id;
    var import0, import1, import2, import4, CommonModuleInjector, CommonModuleNgFactory;
    return {
        setters: [
            function (import0_1) {
                import0 = import0_1;
            },
            function (import1_1) {
                import1 = import1_1;
            },
            function (import2_1) {
                import2 = import2_1;
            },
            function (import4_1) {
                import4 = import4_1;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            CommonModuleInjector = (function (_super) {
                __extends(CommonModuleInjector, _super);
                function CommonModuleInjector(parent) {
                    return _super.call(this, parent, [], []) || this;
                }
                Object.defineProperty(CommonModuleInjector.prototype, "_NgLocalization_1", {
                    get: function () {
                        if ((this.__NgLocalization_1 == null)) {
                            (this.__NgLocalization_1 = new import2.NgLocaleLocalization(this.parent.get(import4.LOCALE_ID)));
                        }
                        return this.__NgLocalization_1;
                    },
                    enumerable: true,
                    configurable: true
                });
                CommonModuleInjector.prototype.createInternal = function () {
                    this._CommonModule_0 = new import1.CommonModule();
                    return this._CommonModule_0;
                };
                CommonModuleInjector.prototype.getInternal = function (token, notFoundResult) {
                    if ((token === import1.CommonModule)) {
                        return this._CommonModule_0;
                    }
                    if ((token === import2.NgLocalization)) {
                        return this._NgLocalization_1;
                    }
                    return notFoundResult;
                };
                CommonModuleInjector.prototype.destroyInternal = function () {
                };
                return CommonModuleInjector;
            }(import0.NgModuleInjector));
            exports_1("CommonModuleNgFactory", CommonModuleNgFactory = new import0.NgModuleFactory(CommonModuleInjector, import1.CommonModule));
        }
    };
});
